### PR TITLE
Remove ILegacyReactInstance, and react::uwp::IReactInstance

### PR DESCRIPTION
**Pull request by acoates-ms**
This is some more clean up from getting rid of ReactUwp.  This change moves the viewmanagers and a couple of other remaining places using `ILegacyReactInstance` or `react::uwp::IReactInstance` to using `IReactContext` instead.   Then removes `ILegacyReactInstance`, `react::uwp::IReactInstance` and `UwpReactInstanceProxy`

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/6175)

### CODE REVIEW

File: vnext/Microsoft.ReactNative/Modules/AppThemeModuleUwp.cpp [hunk 1]
```diff
@@ -73,9 +74,7 @@ std::string AppTheme::formatRGB(winrt::Windows::UI::Color ElementColor) {
 }
 
 void AppTheme::fireEvent(std::string const &eventName, folly::dynamic &&eventData) {
-  if (auto instance = m_wkReactInstance.lock()) {
-    instance->CallJsFunction("RCTDeviceEventEmitter", "emit", folly::dynamic::array(eventName, std::move(eventData)));
-  }
+  m_context->CallJSFunction("RCTDeviceEventEmitter", "emit", folly::dynamic::array(eventName, std::move(eventData)));
```
Comment (NickGerleman) [inline vnext/Microsoft.ReactNative/Modules/AppThemeModuleUwp.cpp:77]: Just for my own understanding, does the context internally keep week ownership?

**Review by NickGerleman — APPROVED** [review]

File: vnext/Microsoft.ReactNative/Modules/AppThemeModuleUwp.cpp (hunk 1 above)
Comment (acoates-ms) [reply vnext/Microsoft.ReactNative/Modules/AppThemeModuleUwp.cpp:77]: Yes.  Context is basically a weak pointer on an instance.

File: vnext/Microsoft.ReactNative/Views/TextInputViewManager.cpp [hunk 2]
```diff
@@ -267,51 +260,49 @@ void TextInputShadowNode::registerEvents() {
...
-            }
-            instance->DispatchEvent(tag, "topTextInputSubmitEditing", std::move(eventDataSubmitEditing));
+          folly::dynamic eventDataSubmitEditing = {};
+          if (m_isTextBox) {
+            eventDataSubmitEditing = folly::dynamic::object("target", tag)(
+                "text", HstringToDynamic(control.as<xaml::Controls::TextBox>().Text()));
+          } else {
+            eventDataSubmitEditing = folly::dynamic::object("target", tag)(
+                "text", HstringToDynamic(control.as<xaml::Controls::PasswordBox>().Password()));
           }
```
Comment (dannyvv) [inline vnext/Microsoft.ReactNative/Views/TextInputViewManager.cpp:279]: nit: Should we factor this a bit cleaner, both branches only differ by control type, yet share a lot of code together.

```suggestion
          auto text = m_isTextBox 
              ? control.as<xaml::Controls::TextBox>().Text()
              : control.as<xaml::Controls::PasswordBox>().Password();
          eventDataSubmitEditing = folly::dynamic::object("target", tag)(
              "text", HstringToDynamic(text));
```

File: vnext/ReactWindows-Desktop.sln [hunk 3]
```diff
@@ -169,12 +169,12 @@ Global
...
 		{88BAB0FA-E1AC-4DA7-A30C-F91702A8EADB}.Release|x86.ActiveCfg = Release|Win32
 		{88BAB0FA-E1AC-4DA7-A30C-F91702A8EADB}.Release|x86.Build.0 = Release|Win32
-		{96CD24DC-91C2-480A-BC26-EE2250DA80D7}.Debug|ARM64.ActiveCfg = Debug|ARM64
+		{96CD24DC-91C2-480A-BC26-EE2250DA80D7}.Debug|ARM64.ActiveCfg = Debug|Win32
 		{96CD24DC-91C2-480A-BC26-EE2250DA80D7}.Debug|x64.ActiveCfg = Debug|x64
 		{96CD24DC-91C2-480A-BC26-EE2250DA80D7}.Debug|x64.Build.0 = Debug|x64
 		{96CD24DC-91C2-480A-BC26-EE2250DA80D7}.Debug|x86.ActiveCfg = Debug|Win32
 		{96CD24DC-91C2-480A-BC26-EE2250DA80D7}.Debug|x86.Build.0 = Debug|Win32
-		{96CD24DC-91C2-480A-BC26-EE2250DA80D7}.Release|ARM64.ActiveCfg = Release|ARM64
+		{96CD24DC-91C2-480A-BC26-EE2250DA80D7}.Release|ARM64.ActiveCfg = Release|Win32
```
Comment (dannyvv) [inline vnext/ReactWindows-Desktop.sln:177]: Just checking if this is intentional, as ARM64 is mapped to Win32, of if this is just VS having a bad day :P

**Review by dannyvv — APPROVED** [review]

File: vnext/ReactWindows-Desktop.sln (hunk 3 above)
Comment (dannyvv) [reply vnext/ReactWindows-Desktop.sln:177]: note: no vcxproj change suggests this should have happened...

**Comment by vmoroz** [comment]
> #pragma once

Should we remove this file?

---
Refers to: vnext/Microsoft.ReactNative/ReactHost/React_Win.h:4 in ea6263b. [](commit_id = ea6263b08747c2cc7e2b5d6b0857344631233257, deletion_comment = False)

File: vnext/ReactWindows-Desktop.sln (hunk 3 above)
Comment (acoates-ms) [reply vnext/ReactWindows-Desktop.sln:177]: Hmm.. I didn't do anything manually.  This was VS.  These appear to be the test projects that got switched, which I dont think build in arm64.  So this is probably correct.

**Comment by acoates-ms** [comment]
> > #pragma once
> 
> Should we remove this file?
> 
> Refers to: vnext/Microsoft.ReactNative/ReactHost/React_Win.h:4 in [ea6263b](https://github.com/microsoft/react-native-windows/commit/ea6263b08747c2cc7e2b5d6b0857344631233257). [](commit_id = [ea6263b](https://github.com/microsoft/react-native-windows/commit/ea6263b08747c2cc7e2b5d6b0857344631233257), deletion_comment = False)

There are a couple of headers that are almost empty.  But the PR is already big enough and going through and cleaning up the headers adds a bunch to the churn.  We can do that as a follow up.

**Review by vmoroz — APPROVED** [review]
:shipit:

**Review by NikoAri — APPROVED** [review]
:shipit:

**Comment by QueenTallicia69-cloud** [comment]
DON'T YOU DARE REMOVE ME FROM YOUR MAILING ADDRESS

Op vr 2 okt. 2020 22:46 schreef Andrew Coates <notifications@github.com>:

**Pull request merged**